db_dtypes/__init__.py

```diff
@@ -125,20 +133,20 @@ def _datetime(
             if not parsed:
                 raise ValueError(f"Bad time string: {repr(scalar)}")
 
-            hours = parsed.group("hours")
-            minutes = parsed.group("minutes")
-            seconds = parsed.group("seconds")
+            hour = parsed.group("hours")
+            minute = parsed.group("minutes")
+            second = parsed.group("seconds")
             fraction = parsed.group("fraction")
-            microseconds = int(fraction.ljust(6, "0")[:6]) if fraction else 0
-            return datetime.datetime(
-                1970,
-                1,
-                1,
-                int(hours),
-                int(minutes) if minutes else 0,
-                int(seconds) if seconds else 0,
-                microseconds,
-            )
+            nanosecond = int(fraction.ljust(9, "0")[:9]) if fraction else 0
+            return pandas.Timestamp(
+                year=1970,
+                month=1,
+                day=1,
+                hour=int(hour),
+                minute=int(minute) if minute else 0,
+                second=int(second) if second else 0,
+                nanosecond=nanosecond,
+            ).to_datetime64()
```

Comment:
Towards #19

Reply:
Probably should have been a separate PR, but I noticed `_datetime` had inconsistent types (sometimes pandas.Timestamp, sometime datetime.datetime, sometime numpy.datetime64). We want a `numpy.datetime64` in the end, so I unified it.